report/metrics_file_descriptors.go

```diff
@@ -31,29 +37,49 @@ func SetupLinuxBSDFDMetrics(logger *logp.Logger, reg *monitoring.Registry, proce
 }
 
 func FDUsageReporter(logger *logp.Logger, processStats *process.Stats) func(_ monitoring.Mode, V monitoring.Visitor) {
+	pid, err := process.GetSelfPid(processStats.Hostfs)
+	if err != nil {
+		logger.Error("Error while retrieving pid: %v", err)
+		return nil
```

Comment:
I don't think it's safe to return `nil` here. It'll end up being called latter on without a nil check.

Originally there was always a non-nil return as everything would happen on the callback, on error, the callback would log and do nothing. 

I'm not super familiar with how those metrics are used, so I'm not sure if this callback is called only once per application run or if it might be called several times. If it's the latter, I'm wondering if there could be a situation where `process.GetSelfPid` could eventually succeed.
It's just reading a file, it's unlikely the permissions for the file would change in runtime, the process wouldn't get more permissions in runtime. SO I believe handling it before creating the callback might be ok. However it should always return a valid callback, no-op on error.

Let me try to find someone who knows better this part of this lib

Reply:
yeah. It doesn't look safe.

Here's my understanding of this part of the lib:
1. We initialise all the registries in beats.
2. If monitoring is enabled, we log the internal metrics for each registry every 30s by default.
3. In this case, we have setup a callback here https://github.com/elastic/elastic-agent-system-metrics/blob/c33dfcd19bc9991336bbb406f5889bbc99ccb822/report/metrics_file_descriptors.go#L35-L37
   - `FDUsageReporter` returns a function, which is the one responsible for capturing metrics. It will be called when "visit" this registry key. [Here's](https://github.com/elastic/elastic-agent-libs/blob/b06f9b2ef69e8d4007e4113af4dea928450b7646/monitoring/metrics.go#L235) the code responsible for calling it 
   - As you can see, there is no null check in place and I suspect it will create a panic if we return `nil`.
 
 
I suggest we return an empty function 

```suggestion
		return return func(_ monitoring.Mode, V monitoring.Visitor) {
		      	 V.OnRegistryStart()
		          V.OnRegistryFinished()
		}
```

Reply:
> I'm not super familiar with how those metrics are used, so I'm not sure if this callback is called only once per application run or if it might be called several times

It will be called as per configured period. The key `logging.metrics.period` in beat configuration is the one that controls the frequency I guess.